constraint/pkg/client/client.go

```diff
@@ -214,19 +220,43 @@ func createTemplatePath(target, name string) string {
 
 // CreateCRD creates a CRD from template
 func (c *client) CreateCRD(ctx context.Context, templ *v1alpha1.ConstraintTemplate) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
-	if err := validateTargets(templ); err != nil {
+	nm := templ.ObjectMeta.Name
+	specCRD := &templ.Spec.CRD
+	targets := templ.Spec.Targets
+	if err := c.validateGenericCRD(nm, specCRD, targets); err != nil {
+		return nil, err
+	}
+	return c.createGenericCRD(nm, specCRD, targets, constraintGroup)
+}
+
+func (c *client) CreateMutationCRD(ctx context.Context, templ *v1alpha1.MutationTemplate) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
```

Comment:
It's probably better to have CreateCRD() take an interface{} typed variable and do type switching

Reply:
Cam, FYI: https://tour.golang.org/methods/16

Reply:
Per a conversation with Cam, we can consolidate this into one CreateCRD function later. This would be a requirement for merging into master.